### PR TITLE
Fix/988/pdosworkchain

### LINKED ISSUE #988: PdosWorkChain except when tot_magnetization is used
The PdosWorkChain except when dealing with a calculation were `tot_magnetization` is considered.
The issue is from this line `self.ctx.nscf_fermi = workchain.outputs.output_parameters.dict.fermi_energy`

When using tot_magnetization there are two energies `fermi_energy_up` and `fermi_energy_down` instead of fermi_energy. 

This affects the rest of the logic of the workchain that only considers fermi_energy

### PR DESCRIPTION
**Pull request by AndresOrtegaGuerrero**
Fixes #988

### CODE REVIEW

**Review by mbercx — APPROVED** [review]
Thanks @AndresOrtegaGuerrero!

File: tests/conftest.py [hunk 1]
```diff
@@ -51,14 +51,17 @@ def fixture_code(fixture_localhost):
...
 
+        query = QueryBuilder().append(
+            InstalledCode,
+            filters={'label': label},
+        )
         try:
-            return load_code(label=label)
-        except exceptions.NotExistent:
+            return query.first()[0]
+        except TypeError:
```
Comment (sphuber) [inline tests/conftest.py:64]: Why was this changed? @mbercx I don't understand this nor do I think it is correct. I would revert this

Comment (mbercx) [reply tests/conftest.py:64]: I'm not 100% what was happening under the `pytest` hood, but when using `pytest.mark.parametrize` to test the two possible NSCF output parameters, it seems to somehow have created two instances of the `quantumespresso.pw` code, and the second test fails with:

```
aiida.common.exceptions.MultipleObjectsError: multiple Code entries found with LABEL<test.quantumespresso.pw>
```

My solution is a bit of a late-night hack, I'll admit. ^^ 

Looking into this further, it actually seems that VSCode running the test in parallel causes the issue. After reverting reverting the change [here](https://github.com/mbercx/aiida-quantumespresso/commit/b23e5eca84cd2a20b63c651d05955debd90f4063), the tests still pass fine. Running `pytest` manually locally also works without issue.

Happy to open a PR to revert, but maybe there is also a more elegant solution to have the tests run properly in parallel?

Comment (sphuber) [reply tests/conftest.py:64]: Well, the tests are not designed to be run in parallel, especially not if they use the same storage backend. Even if you "fix" this case, there is bound to be a shitload of other problems lying in wait when you have multiple tests run against the same database. So I really think you just shouldn't run them in parallel.

Comment (mbercx) [reply tests/conftest.py:64]: Opened https://github.com/aiidateam/aiida-quantumespresso/pull/995 to revert.
